src/SSAS.NET/StratisId.cs

```diff
@@ -1,134 +1,155 @@
-
-        /// <inheritdoc />
-        public override bool Equals(object obj) => obj is StratisId other && Equals(other);
-
-        /// <inheritdoc />
-        public static bool operator ==(StratisId a, StratisId b) => a.Equals(b);
-
-        /// <inheritdoc />
-        public static bool operator !=(StratisId a, StratisId b) => !a.Equals(b);
-
-        /// <summary>
-        /// Converts the string representation of a Stratis ID URI or callback to its <see cref="StratisId" /> equivalent. A return value indicates whether the operation succeeded.
-        /// </summary>
-        /// <param name="value">A string containing the Stratis ID URI or callback to convert.</param>
-        /// <param name="stratisId">When this method returns true, contains the <see cref="StratisId" /> equivalent of the value, or null if the conversion failed.</param>
-        /// <returns>true if value was converted successfully; otherwise, false.</returns>
-        public static bool TryParse(string value, out StratisId stratisId)
-        {
-            stratisId = null;
-
-            var callback = value.StartsWith(SchemeWithDelimeter) ? value[SchemeWithDelimeter.Length..] : value;
-            if (callback.StartsWith("//")) return false;
-
-            var callbackParts = callback.Split('?');
-            if (callbackParts.Length != 2) return false;
-
-            var queryString = callbackParts[1];
-            if (queryString is null) return false;
-
-            var queryParams = new Dictionary<string, string>();
-
-            var queryParts = queryString.Split('&');
-            foreach (var part in queryParts)
-            {
-                var equalSignIndex = part.IndexOf('=');
-                if (equalSignIndex == -1) continue;
-
-                queryParams.Add(part[..equalSignIndex], part[(equalSignIndex + 1)..]);
-            }
-
-            if (!queryParams.ContainsKey(UidKey) || string.IsNullOrWhiteSpace(queryParams[UidKey])) return false;
-
-            long? exp = null;
-            if (queryParams.ContainsKey(ExpKey))
-            {
-                if (!long.TryParse(queryParams[ExpKey], out var expiry)) return false;
-                exp = expiry;
-            }
-
-            stratisId = new StratisId(callbackParts[0], queryParams[UidKey], exp);
-
-            return true;
-        }
-    }
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SSAS.NET
+{
+    /// <summary>
+    /// A Stratis ID URI defined in Stratis Signature Auth Specification.
+    /// </summary>
+    public class StratisId : IEquatable<StratisId>
+    {
+        /// <summary>
+        /// Stratis ID URI scheme.
+        /// </summary>
+        public const string Scheme = "sid";
+
+        private const string UidKey = "uid";
+        private const string ExpKey = "exp";
+
+        private const string SchemeWithDelimeter = "sid:";
+        private const string ProtocolHandlerSchemeWithDelimeter = "web+sid://";
```

Comment:
Need to go through the PR but wondering if we really _need_ to include this in here, as soon as this is implemented we need to support different get/post requests etc rather than clients just handling it.

Reply:
Why would that be? It uses the same flow. This simply adds support for parsing a protocol handler and converting to a protocol URI string. Could be used for a Razor/Blazor dApp, for example